AAInstrumentation/core.py

```diff
@@ -423,8 +515,29 @@ def get_n_random_sequences(self, num_candidates: int, num_runs: int) -> list[tup
 
     count_per_file = driver.get_candidates_per_file(files)
 
-    print(count_per_file)
     for file in files:
-        driver.run_step_single_func(file, 0, [1, 2, 3])
-        file_name = initial_dir / file
-        print(file_name, driver.assemble_and_measure_file(file_name))
+        queries = driver.get_queries_per_pass(file, [0, 1, 2])
+        relaxed_queries = driver.get_queries_per_pass(file)
+        if all(
+            [
+                relaxed_queries[pass_name]["MayAlias"] == queries[pass_name]["MayAlias"]
+                for pass_name in relaxed_queries.keys()
+            ]
+            + [
+                relaxed_queries[pass_name]["PartialAlias"]
+                == queries[pass_name]["PartialAlias"]
+                for pass_name in relaxed_queries.keys()
+            ]
+            + [
+                relaxed_queries[pass_name]["MustAlias"]
+                == queries[pass_name]["MustAlias"]
+                for pass_name in relaxed_queries.keys()
+            ]
+            + [
+                relaxed_queries[pass_name]["NoAlias"] == queries[pass_name]["NoAlias"]
+                for pass_name in relaxed_queries.keys()
+            ]
```

Comment:
There's a lot of repeated code here, can you factor it out to a single function to make the code more readable?